src/components/modules/epics/ContributionsEpicReminder.tsx

```diff
@@ -1,288 +1,86 @@
-                                <Button
-                                    iconSide="right"
-                                    icon={<SvgArrowRightStraight />}
-                                    type="submit"
-                                    disabled={isSubmittingState}
-                                    css={getCustomSubmitStyles(isSubmittingState)}
-                                >
-                                    Set a reminder
-                                </Button>
-                            </div>
-                        </>
-                    )}
-
-                    {isErrorState && (
-                        <p css={errorTextStyles}>
-                            Sorry we couldn&apos;t set a reminder for you this time. Please try
-                            again later.
-                        </p>
-                    )}
-                </form>
-
-                {!isSuccessState && (
-                    <p css={formTextStyles}>
-                        We will send you a maximum of two emails {reminderDateWithPreposition}. To
-                        find out what personal data we collect and how we use it, view our{' '}
-                        <a
-                            css={linkStyles}
-                            href="https://www.theguardian.com/help/privacy-policy"
-                            target="_blank"
-                            rel="noopener noreferrer"
-                        >
-                            Privacy Policy
-                        </a>
-                        .
-                    </p>
-                )}
-
-                {isSuccessState && (
-                    <>
-                        <h4 css={remindHeading}>Thank you! Your reminder is set.</h4>
-                        <p css={successTextStyles}>
-                            We will be in touch to invite you to contribute. Look out for a message
-                            in your inbox {reminderDateWithPreposition}. If you have any questions
-                            about contributing, please{' '}
-                            <a href="mailto:contribution.support@theguardian.com" css={linkStyles}>
-                                contact us
-                            </a>
-                            .
-                        </p>
-                    </>
-                )}
-            </div>
-        </div>
+    return initialEmailAddress ? (
+        <ContributionsEpicReminderSignedIn
+            reminderLabel={reminderLabel}
+            reminderStatus={reminderStatus}
+            onSetReminderClick={() => setReminder(initialEmailAddress)}
+            onCloseReminderClick={onCloseReminderClick}
+        />
+    ) : (
+        <ContributionsEpicReminderSignedOut
+            reminderLabel={reminderLabel}
+            reminderStatus={reminderStatus}
+            onSetReminderClick={setReminder}
+            onCloseReminderClick={onCloseReminderClick}
+        />
```

Comment:
I mentioned in the description, but just to reiterate: making these much more presentational (e.g. by doing the fetch call in the parent component, and then passing the status down) made for a much better development flow!

Reply:
💯